### PR TITLE
Carnival Update

### PR DESCRIPTION
**Pull request by EnderMage99**
## About The Pull Request

Carnival Updates:
1. Adds 3 new silk types and gives all silks their masterpiece/elegant variants.
2. Updates all of the armor recipes to be a lot more equal. So each one takes about the same amount of effort. Also, Now the recipes are more complicated in which silk they use.
3. Lets the Carnival create Human silk by butchering Lovetown and lets them craft Human Silk, (However it does take a bit of effort.)
4. Carnival can no longer Bucher dead humans which have a soul.
5. Makes their spear deal half damage to humans.
6. Carnival can now spawn in City Fixers
7. Adds a new book, which lets the carnival create some really powerful endgame armor.
8. Gives them a "Silk Bag" that is able to hold a large amount of silk without taking up a lot of room.
9. Silk can only be applied to armor by the Carnival, and only the Carnival can read their weaving books.
10. It organizes the Carnival crafting recipes which makes their code much cleaner.

Also, This adds the "Meat Bag" for fixers.
This bag can be bought at fixer vending machines and it can hold around 12 different meat slabs at one time.
It should make selling/carrying all of that meat more convenient for fixers.

## Why It's Good For The Game

Along with adding a bit more depth to the carnival by encouraging them to combine their silk into better types to make armor, this helps City Fixers with making armor much more accessible there.

This also makes the Carnival much less antagonist towards humans, which should allow them to focus on the selling aspect of them, rather than the hunting humans part.

## Changelog
:cl:
add: added Shrimple, Crimson and Violet Silk
reworked: reworked all of the recipes for crafting armor to be more equal.
add: the silk/meat bags to CoL/CF
add: the carnival to CF
nerf: the carnival spear deals half damage to humans
add: makes silk stuff exclusive with the carnival
add: the carnival is now able to craft midnight ordeal armor, at a VERY high cost.
nerf: the silkweaver can no longer bucher humans with souls.
/:cl:

### CODE REVIEW

**Comment by Gboster-0** [comment]
wow, quite a huge balance rework

**Comment by Kitsunemitsu** [comment]
I don't really want to add bloat to more CF roles, especially one that's "hostile" by default like Carnivals.

I'd like it instead to be attached to the workshop, to give them more to do.

**Comment by EnderMage99** [comment]
> I don't really want to add bloat to more CF roles, especially one that's "hostile" by default like Carnivals.
> 
> I'd like it instead to be attached to the workshop, to give them more to do.

Well, I trying to make them less hostile to players, since they now deal less damage to humans and can’t butcher dead people.

 But, I like the idea of adding them to the workshop since there a lot less players in CF.

However, I am also wondering how should I handle the fact that the carnival can go into the backstreets but the workshop can’t? Should I give them the carnival spear or not?

**Comment by EnderMage99** [comment]
> I don't really want to add bloat to more CF roles, especially one that's "hostile" by default like Carnivals.
> 
> I'd like it instead to be attached to the workshop, to give them more to do.

Also, When you say “don't really want to add bloat to more CF roles” do you mean you don’t want me to add more silk types?

**Comment by Kitsunemitsu** [comment]
> > I don't really want to add bloat to more CF roles, especially one that's "hostile" by default like Carnivals.
> > I'd like it instead to be attached to the workshop, to give them more to do.
> 
> Also, When you say “don't really want to add bloat to more CF roles” do you mean you don’t want me to add more silk types?

Actually It's more that Workshop is already bored as hell by midround; I think that giving them armor crafting (and maybe prosthetics too soon, if there's the space near their workshop) would help them be more occupied into the lategame

**Comment by Kitsunemitsu** [comment]
> However, I am also wondering how should I handle the fact that the carnival can go into the backstreets but the workshop can’t? Should I give them the carnival spear or not?

About them going into the backstreets, first off, they can, and I want to encourage them to, and second off, they can get the bodies shipped straight to them if needed.

Just reskin the carnival spear as a subtype or whatever if needed.

**Comment by Tokanova** [comment]
> > > I don't really want to add bloat to more CF roles, especially one that's "hostile" by default like Carnivals.
> > > I'd like it instead to be attached to the workshop, to give them more to do.
> > 
> > 
> > Also, When you say “don't really want to add bloat to more CF roles” do you mean you don’t want me to add more silk types?
> 
> Actually It's more that Workshop is already bored as hell by midround; I think that giving them armor crafting (and maybe prosthetics too soon, if there's the space near their workshop) would help them be more occupied into the lategame

Can the workshop not be made 'taller', so it goes further into the backstreets? That'd give more room.

**Comment by EnderMage99** [comment]
> > > > I don't really want to add bloat to more CF roles, especially one that's "hostile" by default like Carnivals.
> > > > I'd like it instead to be attached to the workshop, to give them more to do.
> > > 
> > > 
> > > Also, When you say “don't really want to add bloat to more CF roles” do you mean you don’t want me to add more silk types?
> > 
> > 
> > Actually It's more that Workshop is already bored as hell by midround; I think that giving them armor crafting (and maybe prosthetics too soon, if there's the space near their workshop) would help them be more occupied into the lategame
> 
> Can the workshop not be made 'taller', so it goes further into the backstreets? That'd give more room.

Sadly, That is where the backstreets load, so it would get overwritten by the backstreets spawner.

File: ModularTegustation/ego_weapons/melee/non_abnormality/carnival.dm [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	desc = "A spear that the Carnival uses to hunt down their prey."
 	icon_state = "carnival_spear"
 	inhand_icon_state = "carnival_spear"
-	special = "Deal double damage to mobs of the backstreets."
+	special = "Deal double damage to mobs of the backstreets, But deal half damage to humans."
```
Comment (Gboster-0) [inline ModularTegustation/ego_weapons/melee/non_abnormality/carnival.dm:6]: This is no longer accurate yeah?
```suggestion
	special = "Deal double damage to mobs of the backstreets."
```

File: code/modules/jobs/job_types/city/carnival.dm [hunk 2]
```diff
@@ -30,9 +30,10 @@
 	ADD_TRAIT(H, TRAIT_WORK_FORBIDDEN, JOB_TRAIT)
 	ADD_TRAIT(H, TRAIT_COMBATFEAR_IMMUNE, JOB_TRAIT)
 	H.set_species(/datum/species/synth)
-	job_important = "If there is an L-Corp facility nearby, do not enter it. Fixers are not inherently hostile to you, but they can find a reason to put you down. \
+	job_important = "You are alowed to enter the backstreets to hunt for silk. However, DO NOT LOOT THE WEAPONS, CASH OR ARMOR FROM THE BACKSTREETS! \
```
Comment (Gboster-0) [inline code/modules/jobs/job_types/city/carnival.dm:33]:
```suggestion
	job_important = "You are allowed to enter the backstreets to hunt for silk. However, DO NOT LOOT THE WEAPONS, CASH OR ARMOR FROM THE BACKSTREETS! \
```

File: ModularTegustation/ego_weapons/melee/non_abnormality/carnival.dm [hunk 3]
```diff
@@ -36,3 +36,9 @@
 		force *= 2
 	..()
 	force = initial_force
+
+/obj/item/ego_weapon/city/carnival_spear/weak
+	name = "Worn-down Carnival Spear"
+	force = 22
+	attribute_requirements = list(
+	)
```
Comment (Gboster-0) [inline ModularTegustation/ego_weapons/melee/non_abnormality/carnival.dm:44]:
```suggestion
	attribute_requirements = list()
```

File: code/datums/components/crafting/recipes/carnival_silk/armor_weaving/carnival_blade_lineage.dm [hunk 4]
```diff
@@ -0,0 +1,33 @@
+//Blade Lineage Recipes
+/datum/crafting_recipe/blade_lineage_salsu
+	name = "Blade Lineage Salsu Robe"
+	result = /obj/item/clothing/suit/armor/ego_gear/city/blade_lineage_salsu
+	reqs = list(/obj/item/stack/sheet/silk/steel_simple = 3,
+				/obj/item/stack/sheet/silk/amber_advanced = 1,
+				/obj/item/stack/sheet/silk/human_simple = 1)
```
Comment (Gboster-0) [inline code/datums/components/crafting/recipes/carnival_silk/armor_weaving/carnival_blade_lineage.dm:7]: While we do have quite a lot of shitcoded lists, please dont contribute towards them
If the list has 2-3 short items, put it in a single line
if it has 2+ fairly long items, put it in a multi-line like this. (each object being on its own line, this is better for future github additions, removals and changes)
```suggestion
	reqs = list(
		/obj/item/stack/sheet/silk/steel_simple = 3,
		/obj/item/stack/sheet/silk/amber_advanced = 1,
		/obj/item/stack/sheet/silk/human_simple = 1,
	)
```

Comment (EnderMage99) [reply code/datums/components/crafting/recipes/carnival_silk/armor_weaving/carnival_blade_lineage.dm:7]: Oh, I guess I didn't really notice how shitty it looked. I will double check all of my lists today.

Comment (EnderMage99) [reply code/datums/components/crafting/recipes/carnival_silk/armor_weaving/carnival_blade_lineage.dm]: All of them should be fixed now.

**Comment by Kitsunemitsu** [comment]
Requesting removal of Meat bag, It's not intended for players to hold that much meat to encourage returning to the city.

**Comment by EnderMage99** [comment]
> Requesting removal of Meat bag, It's not intended for players to hold that much meat to encourage returning to the city.

Got it, I will just remove it from the vending machine so players can't get it but admins could use it to help out hana.

**Review by Boredone — CHANGES_REQUESTED** [review]

**Comment by Kitsunemitsu** [comment]
Was all the bugs and issues with this fixed?

**Comment by EnderMage99** [comment]
> Was all the bugs and issues with this fixed?

Yep, all of the reported bugs have been fixed.

**Review by Kitsunemitsu — APPROVED** [review]
We've TMed this multiple times, and it looks good to me. Hasn't caused any game breaking, or balance breaking bugs.

**Pull request merged**